main.nf

```diff
@@ -3,219 +3,277 @@
+	                    .unique()
+	// parse txt file with dataset accessions into a queue channel; build combinations with matrix types 
+	IMPORT_PARAMS = Channel
+                .fromPath(params.data_import.training_datasets)
+                .splitCsv(header:false, sep:",")
+                .combine(UNIQUE_MATRIX_TYPES)
+    
 process fetch_query_data{
-	publishDir "${baseDir}/data", mode: 'copy'
+	//publishDir "${params.data_dir}", mode: 'copy'
 	conda "${baseDir}/envs/load_data.yaml"
 	errorStrategy { task.exitStatus == 130 || task.exitStatus == 137  ? 'retry' : 'finish' }   
 	maxRetries 5
 	memory { 16.GB * task.attempt }
+	input:
+	tuple val(dataset_id), val(seq_method), val(num_clust), val(barcode_col), val(cell_type_col), val(matrix_type) from IMPORT_PARAMS
 	
 	output:
-	file("${query_data}/query_10x_data") into QUERY_10X_DIR
-	file("${query_data}/query_sdrf.txt") into CONDENSED_SDRF
-	file("${query_data}/query_${query_markers}") into MARKERS
+	tuple file(dataset_id), val(dataset_id), val(barcode_col), val(cell_type_col), val(matrix_type), val(num_clust) into FETCH_DATA
+        //val(num_clust) into N_CLUST
 	"""
-	get_experiment_data.R\
-		--accesssion-code ${params.download_data.query_acc_code}\
-		--expr-data-type ${params.download_data.expr_data_type}\
-		--normalisation-method ${params.download_data.normalisation_method}\
-		--output-dir-name ${params.download_data.query_output_dir}\
-		--get-condensed-sdrf\
-		--get-marker-genes\
-		--number-of-clusters ${params.download_data.query_num_clust}
-	# rename files to avoid name collisions in subsequent processes
-	mv ${query_data}/10x_data ${query_data}/query_10x_data
-	mv ${query_data}/condensed-sdrf.tsv ${query_data}/query_sdrf.txt
-	mv ${query_data}/${query_markers} ${query_data}/query_${query_markers}
+	if [ ${seq_method} ==  "droplet" ]; then
+        MATRIX_TYPE_UPD="CPM"
+    	else
+        MATRIX_TYPE_UPD=${matrix_type}
+    	fi
+    	get_experiment_data.R\
+                --accesssion-code ${dataset_id}\
+                --config-file ${params.data_import.scxa_import_config_file}\
+             	--matrix-type CPM\
+                --output-dir-name ${dataset_id}\
+                --get-sdrf ${params.data_import.get_sdrf}\
+                --get-condensed-sdrf ${params.data_import.get_cond_sdrf}\
+                --get-idf ${params.data_import.get_idf}\
+                --get-marker-genes ${params.data_import.get_marker_genes}\
+                --number-of-clusters ${num_clust}
 	"""
     }
-  
-	DATASET_ID = Channel.from(params.download_data.query_acc_code).first() 
 }else{
-	QUERY_10X_DIR = Channel.fromPath(params.data_10X, checkIfExists: true).first()
-	CONDENSED_SDRF = Channel.fromPath(params.condensed_sdrf, checkIfExists: true).first()
-	MARKERS = Channel.fromPath(params.markers_metadata, checkIfExists: true).first()
-	DATASET_ID = Channel.from(params.dataset_id) 
+	//QUERY_10X_DIR = Channel.fromPath(params.data_10X, checkIfExists: true).first()
```

Comment:
If you don't need this else could you remove it please? We don't need commented lines in prod code